tf_encrypted/__init__.py

```diff
@@ -24,14 +23,30 @@
 from . import queue
 
 
-_all_prot_funcs = protocol.get_all_funcs()
+__protocol__ = None
+__all_prot_funcs__ = protocol.get_all_funcs()
 
 
 def _prot_func_not_implemented(*args: Any, **kwargs: Any) -> None:
   msg = "This function is not implemented in protocol {}"
   raise Exception(msg.format(inspect.stack()[1][3]))
 
 
+def _update_protocol(prot):
```

Comment:
Mainly because it used to live in protocol.py, but also because it's another bit of global state that might be good to know about on it's own -- happy to either change the name or move it inline though, no strong preference